app/src/main/java/com/example/drawingApp/ColorCircle.kt

```diff
@@ -0,0 +1,106 @@
+package com.example.drawingApp
+
+import android.annotation.SuppressLint
+import android.content.Context
+import android.graphics.*
+import android.util.AttributeSet
+import android.util.Log
+import android.view.MotionEvent
+import android.view.View
+import androidx.core.graphics.toColor
+
+
+@SuppressLint("ClickableViewAccessibility")
+class ColorCircle @JvmOverloads constructor(
+    context: Context,
+    attrs: AttributeSet? = null,
+    defStyle: Int = 0
+) : View(context,attrs,defStyle) {
+    /*
+    what does this custom view do?
+        it is a solid circle
+        when clicked it cycles forward in an order of colors
+        Black, Red, Green, Blue is the order
+     */
+    //variables that should be available
+    /*
+        private state variable
+     */
+    private var state = ColorState.BLACK
+    private var paint = Paint()
+    lateinit var passingColorCircle: (Int) -> Unit
+    //functions
+    /*
+    function that is passed a color, updates the color and the paint
+
+    init with touch listeners
+        on down
+            toggle a held variable
+        on cancel / up toggle off held variable
+            cycle color
+            held = false
+            call lambda to set color to colorPrimary in the viewmodel
+
+    cycler func, cycles forward the color in the order
+        switch statement on the state variable
+
+     */
+
+    init {
+        setOnTouchListener { _, mE ->
+            when (mE.action) {
+                MotionEvent.ACTION_DOWN -> {
+                    cycle()
+                    if(::passingColorCircle.isInitialized)passingColorCircle(state.c)
+                    Log.v("Hamdan","pressed down")
+                }
+                MotionEvent.ACTION_UP, MotionEvent.ACTION_CANCEL -> {
+                    invalidate()
+                }
+            }
+            true
+        }
+    }
+    enum class ColorState(val c: Int){
+        BLACK(Color.parseColor("#000000")),
```

Comment:
I found when working on another project that working with the integers for the colors made things a little easier, since I don't know the integer value of the color off the top of my head I thought just parseing the color should be fine

Reply:
I see what you mean, Color.Black is just already an integer, using an enum is not really needed.